lib/Perl/Critic/Policy/Subroutines/ProhibitUnusedPrivateSubroutines.pm

```diff
@@ -75,6 +82,18 @@ sub _parse_private_name_regex {
 sub violates {
     my ( $self, $elem, $document ) = @_;
 
+    # if we're skipping, return right away
+    $self->{skipping} and return;
+
+    # A "use" not a sub declaration?
+    if ($elem->isa('PPI::Statement::Include')) {
+        # see if we're using something that indicates we shoud skip from now on
+        if ($self->{_skip_when_using}{ $elem->module }) {
+            $self->{skipping} = 1;
+        }
+        return;
+    }
+
```

Comment:
Actually, we already have `Perl::Critic::Document::uses_module()`. Internally it caches stuff so no extra optimization is needed. So just do this:

```
my @skip_modules = keys %{ $self->_skip_when_using }; 
return if any { $document->uses_module($_) } @skip_modules;
```